### DISCUSSION TITLE
New <Head> component: Avoid defining screen titles twice

### DISCUSSION THREAD

**Discussion by cseelus** [expo/router#770] [General]
The addition of the new <Head> component, together with [disabling setting the document.title with React Navigation](https://github.com/expo/router/pull/693), seems to lead to some duplication, since one now would have to define all screen titles twice, once for native via layouts or `<Stack.Screen title>` and once for web via the `<Head>` component?

Is this configurable like the corresponding [React Navigation option](https://reactnavigation.org/docs/navigation-container/#documenttitleenabled)?

If not, one way to get a HTML title, for new projects and when upgrade existing Expo Router v1 apps, without having to add a `<Head>` component to each screen, would be to add it to the main web layout component like this:

```javascript
const screenTitle = // Generate from usePathname() or get Stack.Screen title from layout (how?)
…

<Head>
    <title>{screenTitle} | My App</title>
</Head>
```

Is there a way to get the title of the current screen in a parent layout?

**Comment by EvanBacon** [comment]
Technically head is defining invisible metadata for the "current context" regardless of platform. We could expose the head metadata in the future so custom views could use it as a default, but this would probably not be a great default. As you pointed out, head titles can often be long and contain breadcrumbs like `Evan | Bacon Blog` whereas the on-screen title would look something like `Evan`, meaning there's room for user-space opinions that may not fit perfectly in upstream expo-router/head.

**Comment by cseelus** [comment]
Indeed. What I was trying to achieve, is to reuse the native screen titles for the `<Head>` titles though.

Since every screen usually has a title already defined via `<Screen`> options for native platforms, it feels redundant to additionally define `<Head>` titles for each screen. 

Using something like the following in the main layout component, would make things easier:

```javascript
const screenTitle = // Grab the current Screen title
…

<Head>
    <title>{screenTitle} | My App</title>
</Head>
```
